src/app/gene-browser/gene-browser.component.ts

```diff
@@ -67,7 +68,15 @@ export class GeneBrowserComponent implements OnInit, OnDestroy {
     private geneService: GeneService,
     private datasetsService: DatasetsService,
     private loadingService: FullscreenLoadingService,
-  ) { }
+    private router: Router
+  ) {
+    this.router.events.pipe(
+      filter(event => event instanceof NavigationStart)
+    ).subscribe(() => {
+      this.queryService.cancelStreamPost();
+      this.loadingService.setLoadingStop();
+    });
```

Comment:
It would be good to unsubscribe from this in ngOnDestory. See the `this.subscriptions` variable which is used for this purpose.